### PR TITLE
[FIX] Moved merging check from purchase order level to line level.

### PR DESCRIPTION
**Pull request by n17**
- Link to Avoin Systems pull request: https://github.com/avoinsystems/account-analytic/pull/1

### CODE REVIEW

**Comment by n17** [comment]
I have indeed signed the CLA.

**Comment by aisopuro** [comment]
@pedrobaeza I understand there was at some point some kind of bug with the CLA-bot which caused it to complain about signing the CLA even though it had been signed. Has that issue been fixed? We're pretty sure that @n17 has signed the CLA.

**Comment by pedrobaeza** [comment]
Well, the check was probably done before we inserted the CLA tag in the proper contact (which is handled manually), so don't worry too much about that now. If you do another PR and the message appears again, then please report back.

**Comment by aisopuro** [comment]
This one is still looking for reviewers. Would @jbeficent know if anyone is able?

File: procurement_mto_analytic/models/purchase_order_line.py
Comment (Nikul-Chaudhary) [inline procurement_mto_analytic/models/purchase_order_line.py]: @n17 Use like this `if not self.account_analytic_id:`, not checked len of account

File: procurement_mto_analytic/tests/test_purchase_procurement_analytic.py
Comment (Nikul-Chaudhary) [inline procurement_mto_analytic/tests/test_purchase_procurement_analytic.py]: ` cls.env['account.analytic.account']` Use multiple time, so use like global

File: procurement_mto_analytic/tests/test_purchase_procurement_analytic.py
Comment (Nikul-Chaudhary) [inline procurement_mto_analytic/tests/test_purchase_procurement_analytic.py]: same as `self.env['sale.order.line']`

**Review by Nikul-Chaudhary — COMMENTED** [review]
Improve Code

**Comment by n17** [comment]
@Nikul-Chaudhary Is this what you had in mind?

**Comment by aisopuro** [comment]
@Nikul-Chaudhary is this ready to merge now or is something still needed?

**Review by Nikul-Chaudhary — APPROVED** [review]
Code Review LGTM :+1:

**Comment by aisopuro** [comment]
Thanks @Nikul-Chaudhary . Do you know anyone who would be able to do the second review and merge if OK?

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza please take a look?

**Comment by pedrobaeza** [comment]
Please explain the purpose of the PR

**Comment by Nikul-Chaudhary** [comment]
> Please explain the purpose of the PR

can you please @n17 n17 @aisopuro

**Comment by n17** [comment]
@pedrobaeza
This PR changes the logic by which `procurement_mto_analytic` decides if new lines should be merged into existing purchase orders. The original ported functionality did not merge order lines to the same purchase if they had different analytic accounts, even if they were otherwise compatible (same supplier etc). This code changes the functionality so that the different lines are added to the same purchase order, even though they stay separate on the purchase order. This way the module can make use of procurement settings that allow procured amounts to be collected onto larger purchase orders, instead of always forcing them to be split into many smaller orders.

File: procurement_mto_analytic/models/purchase_order_line.py
Comment (pedrobaeza) [inline procurement_mto_analytic/models/purchase_order_line.py]: Do directly: `if values.get('account_analytic_id') != self.account_analytic_id.id:`, remove previous `if`

File: procurement_mto_analytic/models/purchase_order_line.py
Comment (pedrobaeza) [inline procurement_mto_analytic/models/purchase_order_line.py]: Do it a docstring (with `"""`)

File: procurement_mto_analytic/models/purchase_order_line.py
Comment (pedrobaeza) [inline procurement_mto_analytic/models/purchase_order_line.py]: Remove blank lines inside a method. It's a waste of space and doesn't add anything valuable.

File: procurement_mto_analytic/tests/test_purchase_procurement_analytic.py
Comment (pedrobaeza) [inline procurement_mto_analytic/tests/test_purchase_procurement_analytic.py]: Remove empty lines inside a method

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Remove all extra blank lines inside methods.

Add yourself to contributors in the module and put your copyright in the new file

**Comment by n17** [comment]
@pedrobaeza I made the requested changes but the lint check fails with a strange error. Could you please take a look?

**Comment by pedrobaeza** [comment]
The lint error is due to https://github.com/OCA/maintainer-quality-tools/issues/589. Working on fixing it.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
